pkg/storageops/aws/aws.go

```diff
@@ -18,90 +22,46 @@ type ec2Ops struct {
 	mutex    sync.Mutex
 }
 
-const (
-	SetIdentifierNone = "None"
-)
+// ErrAWSEnvNotAvailable is the error type when aws credentails are not set
+var ErrAWSEnvNotAvailable = fmt.Errorf("AWS credentials are not set in environment")
 
-// Custom AWS volume error codes.
-const (
-	_ = iota + 5000
-	ErrVolDetached
-	ErrVolInval
-	ErrVolAttachedOnRemoteNode
-)
+// NewEnvClient creates a new AWS storage ops instance using environment vars
+func NewEnvClient() (storageops.Ops, error) {
+	region := os.Getenv("AWS_REGION")
+	if len(region) == 0 {
+		return nil, ErrAWSEnvNotAvailable
+	}
 
-// StorageError error returned for AWS volumes
-type StorageError struct {
-	// Code is one of AWS volume error codes.
-	Code int
-	// Msg is human understandable error message.
-	Msg string
-	// Instance provides more information on the error.
-	Instance string
-}
+	instance := os.Getenv("AWS_INSTANCE_NAME")
+	if len(instance) == 0 {
+		return nil, ErrAWSEnvNotAvailable
+	}
 
-// StorageOps interface to perform basic operations on aws.
-type StorageOps interface {
-	// Create volume based on input template volume.
-	// Apply labels as tags on EBS volumes
-	Create(template *ec2.Volume, labels map[string]string) (*ec2.Volume, error)
-	// Attach volumeID.
-	// Return attach path.
-	Attach(volumeID string) (string, error)
-	// Detach volumeID.
-	Detach(volumeID string) error
-	// Delete volumeID.
-	Delete(volumeID string) error
-	// Inspect volumes specified by volumeID
-	Inspect(volumeIds []*string) ([]*ec2.Volume, error)
-	// DeviceMappings returns map[local_volume_path]->aws volume ID
-	DeviceMappings() (map[string]string, error)
-	// Enumerate EBS volumes that match given filters.  Organize them into
-	// sets identified by setIdentifier.
-	// labels can be nil, setIdentifier can be empty string.
-	Enumerate(volumeIds []*string,
-		labels map[string]string,
-		setIdentifier string,
-	) (map[string][]*ec2.Volume, error)
-	// DevicePath for attached EBS volume.
-	DevicePath(volume *ec2.Volume) (string, error)
-	// Snapshot EBS volume
-	Snapshot(volumeID string, readonly bool) (*ec2.Snapshot, error)
-	// ApplyTags
-	ApplyTags(v *ec2.Volume, labels map[string]string) error
-	// RemoveTags removes labels from the volume
-	RemoveTags(v *ec2.Volume, labels map[string]string) error
-	// Tags
-	Tags(v *ec2.Volume) map[string]string
-}
+	_, err := credentials.NewEnvCredentials().Get()
+	if err != nil {
```

Comment:
nit: if _, err := credentials.NewEnvCredentials(); err != nil {

Reply:
Fixed in incremental